index.bs

```diff
@@ -1772,6 +1963,278 @@ contexts that have active documents; navigation can also cause contexts to
 </div>
 
+#### The browsingContext.navigationStarted Event #### {#event-browsingContext-navigationStarted}
+
+<dl>
+   <dt>Event Type</dt>
+   <dd>
+      <pre class="cddl local-cddl">
+        BrowsingContextNavigationStartedEvent = {
+         method: "browsingContext.navigationStarted",
+         params: BrowsingContextNavigationInfo
+       }
+      </pre>
+   </dd>
+</dl>
+The [=remote end event trigger=] is:
+
+<div algorithm="remote end event trigger for browsingContext.navigationStarted">
+
+Define the following [=navigation started steps=] with |context| and |navigation status|:
+
+ 1. If the [=current session=] is null, return.
+
+ 1. Let |context id| be the [=browsing context id=] for |context|.
+
+ 1. Let |navigation id| be |navigation status|'s navigation id.
+
+ 1. Let |url| be |navigation status|'s url.
+
+ 1. Let |params| be a [=map=] matching the
+    <code>BrowsingContextNavigationInfo</code> production, with the
+    <code>context</code> field set to |context id|, the <code>navigation</code>
+    field set to |navigation id| and the <code>url</code> field set to the
+    result of the [=URL serializer=] given |url|.
+
+ 1. Let |body| be a [=map=] matching the
+     <code>BrowsingContextNavigationStarted</code> production, with the
+     <code>params</code> field set to |params|.
+
+ 1. Let |related browsing contexts| be a set containing |context|.
+
+ 1. [=Emit an event=] with |body| and |related browsing contexts|.
+
+</div>
+
+#### The browsingContext.fragmentNavigated Event #### {#event-browsingContext-fragmentNavigated}
+
+<dl>
+   <dt>Event Type</dt>
+   <dd>
+      <pre class="cddl local-cddl">
+        BrowsingContextFragmentNavigatedEvent = {
+         method: "browsingContext.fragmentNavigated",
+         params: BrowsingContextFragmentNavigatedParameters
+       }
+
+       BrowsingContextFragmentNavigatedParameters = {
+        context: BrowsingContext,
+        url: text,
+       }
+      </pre>
+   </dd>
+</dl>
+The [=remote end event trigger=] is:
+
+<div algorithm="remote end event trigger for browsingContext.fragmentNavigated">
+
+Define the following [=fragment navigated steps=] with |context| and |navigation status|:
+
+ 1. If the [=current session=] is null, return.
+
+ 1. Let |context id| be the [=browsing context id=] for |context|.
+
+ 1. Let |url| be |navigation status|'s url.
+
+ 1. Let |params| be a [=map=] matching the
+    <code>BrowsingContextFragmentNavigatedParameters</code> production, with the
+    <code>context</code> field set to |context id| and the <code>url</code>
+    field set to the result of the [=URL serializer=] given |url|.
+
+ 1. Let |body| be a [=map=] matching the
+     <code>BrowsingContextFragmentNavigatedEvent</code> production, with the
+     <code>params</code> field set to |params|.
+
+ 1. Let |related browsing contexts| be a set containing |context|.
+
+ 1. [=Emit an event=] with |body| and |related browsing contexts|.
```

Comment:
Same question about ordering here, what will the order of this event and the navigate command finishing be?